src/mon/ConfigMap.cc

```diff
@@ -117,6 +118,101 @@ std::string Section::get_minimal_conf() const
 }
 
 
+// ------------
+
+Profile::Profile(Option *o)
+  : opt(o)
+{
+}
+
+void Profile::dump(Formatter *f) const
+{
+  f->dump_string("level", Option::level_to_str(opt->level));
+  f->dump_string("desc", opt->desc);
+  f->dump_string("fmt_desc", opt->long_desc);
+  f->open_object_section("values");
+  for (auto& i : profile) {
+    f->open_object_section(i.first.c_str());
+    for (auto& j : i.second.options) {
+      f->dump_string(j.first.c_str(), j.second.raw_value);
+    }
+    f->close_section();
+  }
+  f->close_section();
+}
+
+int Profile::parse(
+  CephContext *cct,
+  const std::string& input,
+  std::function<const Option *(const std::string&)> get_opt)
+{
+  json_spirit::mValue json;
+  try {
+    // try json parsing first
+    json_spirit::read_or_throw(input, json);
+    if (json.type() != json_spirit::obj_type) {
+      return -EINVAL;
+    }
+    for (auto& i : json.get_obj()) { // profile_value -> {}
+      if (i.first == "level") {
+	if (i.second.type() != json_spirit::str_type) {
+	  return -EINVAL;
+	}
+	if (i.second.get_str() == "basic") {
```

Comment:
how would you use 'level' for a profile?

Reply:
it's just a queue for the user when documenting the option, and we want something real to show up in the output from ``ceph config get`` (instead of ``unknown``) .  For example, i think ``osd_footprint`` could be a ``basic`` option.